### PR TITLE
Improved password reset and session invalidation for "locked" users

### PR DESCRIPTION
**Pull request by naz**
- Fixed error message when "locked user tries to login to show specific error instead of generic "Access Denied"
- Changed copy for password reset message to: "As a security precaution, your password must be reset. Click \"Forgot?\" to receive an email with instructions."
- Improved error returned by permission layer for non-active users. Allows Ghost-Admin to redirect them to signin screen instead of being stuck with "Resource Not Found" error

### CODE REVIEW

**Comment by naz** [comment]
For reference, old error returned by permission layer for "locked" users was 404 with following content:
```
{
  "errors": [
    {
      "message": "Resource not found error, cannot list tags.",
      "context": "User not found",
      "type": "NotFoundError",
      "details": null,
      "property": null,
      "help": null,
      "code": null,
      "id": "8b4a7890-8dfc-11ea-99a6-1bef62ddfd48"
    }
  ]
}
```

A new response is 401: 
```
{
  "errors": [
    {
      "message": "Authorisation error, cannot list tags.",
      "context": "You are not authorised to make this request.",
      "type": "UnauthorizedError",
      "details": null,
      "property": null,
      "help": null,
      "code": null,
      "id": "474a61f0-8e01-11ea-a648-6ba7d586651b"
    }
  ]
}
```

**Comment by kevinansfield** [comment]
@naz I think the new error response should be using the same `PasswordResetRequiredError` type that we discussed previously so that clients can react accordingly

**Comment by naz** [comment]
@kevinansfield pushed that already :wink:

**Comment by naz** [comment]
Both client and here. The whole PR is ready for review and merge I think

File: core/server/services/permissions/providers.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ module.exports = {
                     }));
                 }
 
+                if (foundUser.get('status') !== 'active') {
```
Comment (naz) [inline core/server/services/permissions/providers.js:18]: Ensures redirect to `/signin` for the Ghost-Admin when the user becomes "locked" with an active session.

File: package.json [hunk 2]
```diff
@@ -83,7 +83,7 @@
     "express-query-boolean": "2.0.0",
     "express-session": "1.17.1",
     "fs-extra": "9.0.0",
-    "ghost-ignition": "4.1.0",
+    "ghost-ignition": "4.2.0",
```
Comment (naz) [inline package.json:86]: Introduces new `UnauthorizedError` error to be used with "locked" users

File: core/server/api/canary/session.js [hunk 3]
```diff
@@ -35,11 +36,23 @@ const session = {
...
+        }).catch(async (err) => {
+            if (!common.errors.utils.isIgnitionError(err)) {
+                throw new common.errors.UnauthorizedError({
+                    message: common.i18n.t('errors.middleware.auth.accessDenied'),
+                    err
+                });
+            }
+
+            if (err.errorType === 'PasswordResetRequiredError') {
+                await api.authentication.generateResetToken({
```
Comment (naz) [inline core/server/api/canary/session.js:48]: @kevinansfield what is your opinion this function call type. It awaits for the response currently but was also thinking to make it "fire-and-forget" style of call. Wdyt?

Comment (kevinansfield) [reply core/server/api/canary/session.js:48]: Only issue with "fire-and-forget" is that problems with mail config never surface, just appears like emails never arrived. That's typically only a problem for self-hosters though.

**Pull request merged**